src/main/scala/gitbucket/core/util/JGitUtil.scala

```diff
@@ -136,32 +144,177 @@ object JGitUtil {
     changeType: ChangeType,
     oldPath: String,
     newPath: String,
-    oldContent: Option[String],
-    newContent: Option[String],
-    oldIsImage: Boolean,
-    newIsImage: Boolean,
+    oldContentInfo: Option[ContentInfo],
+    newContentInfo: Option[ContentInfo],
     oldObjectId: Option[String],
     newObjectId: Option[String],
     oldMode: String,
     newMode: String,
     tooLarge: Boolean,
     patch: Option[String]
-  )
+  ) {
+
+    def getOldEnhancedRenderRequest(
+      branch: String,
+      repository: RepositoryService.RepositoryInfo,
+      enableWikiLink: Boolean,
+      enableRefsLink: Boolean,
+      enableAnchor: Boolean
+    )(implicit context: Context): Option[EnhancedRenderRequest] = {
+      oldContentInfo.map(
+        EnhancedRenderRequest(
+          oldPath.split("/").toList,
+          _,
+          branch,
+          repository,
+          enableWikiLink,
+          enableRefsLink,
+          enableAnchor,
+          context
+        )
+      )
+    }
+
+    def getNewEnhancedRenderRequest(
+      branch: String,
+      repository: RepositoryService.RepositoryInfo,
+      enableWikiLink: Boolean,
+      enableRefsLink: Boolean,
+      enableAnchor: Boolean
+    )(implicit context: Context): Option[EnhancedRenderRequest] = {
+      newContentInfo.map(
+        EnhancedRenderRequest(
+          newPath.split("/").toList,
+          _,
+          branch,
+          repository,
+          enableWikiLink,
+          enableRefsLink,
+          enableAnchor,
+          context
+        )
+      )
+    }
+
+  }
 
   /**
    * The file content data for the file content view of the repository viewer.
    *
-   * @param viewType "image", "large" or "other"
-   * @param size total size of object in bytes
-   * @param content the string content
-   * @param charset the character encoding
+   * @param repository RepositoryInfo
+   * @param commitId Commit id for this file
+   * @param loader ObjectLoader for this file
+   * @param branch branchname for this file
+   * @param filePath path for this file
    */
-  case class ContentInfo(viewType: String, size: Option[Long], content: Option[String], charset: Option[String]) {
+  case class ContentInfo(
+    owner: String,
+    repository: String,
+    commitId: String,
+    loader: ObjectLoader,
+    branch: String,
+    filePath: String
+  ) {
```

Comment:
Basically, case classes shouldn't cause (or depend on) side effect in the constructor. These properties should be set at the outside of case classes (e.g. You can create a factory method in the companion object to do it).

This also makes possible to extract a renderer using pattern matching, not cast of the renderer property.